### PR TITLE
COORD and AREACOORD updated

### LINKED ISSUE #42804: COORD / AREACOORD reports coordinates of held objects as 0,0,0
## Round ID: 
101902 Bagil

## Reproduction:

`AREACOORD(bomb)` used in log_bomber() seems to always point to 0,0,0
Some examples:
```
ADMIN LOG: Denton-30/(Sombrero TX)(?) (FLW) at CentCom Docks (176, 111, 1) (JMP) triggered a PDA explosion on PDA-Xander Neely (Assistant) at Admin Prison (0, 0, 0) (JMP)
ADMIN LOG: CrabCakesMcgee/(Feathers McGraw)(?) (FLW) at Central Primary Hallway (88, 132, 2) (JMP) has primed a classic EMP grenade at Central Primary Hallway (0, 0, 0) (JMP) for detonation.
```

### PR DESCRIPTION
**Pull request by MadmanMartian**
## About The Pull Request
Previously, COORD would return 0,0,0 if it was within an object, now if that is the case, it returns its locations co-ordinates.

closes #42804 

## Why It's Good For The Game

Better logging is always good

## Changelog
Backend changes

### CODE REVIEW

**Comment by AnturK** [comment]
Isn't this macro turf-only ?

File: code/__DEFINES/admin.dm [hunk 1]
```diff
@@ -57,8 +57,8 @@
 #define ADMIN_FULLMONTY_NONAME(user) "[ADMIN_QUE(user)] [ADMIN_PP(user)] [ADMIN_VV(user)] [ADMIN_SM(user)] [ADMIN_FLW(user)] [ADMIN_TP(user)] [ADMIN_INDIVIDUALLOG(user)] [ADMIN_SMITE(user)]"
 #define ADMIN_FULLMONTY(user) "[key_name_admin(user)] [ADMIN_FULLMONTY_NONAME(user)]"
 #define ADMIN_JMP(src) "(<a href='?_src_=holder;[HrefToken(TRUE)];adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)"
-#define COORD(src) "[src ? "([src.x],[src.y],[src.z])" : "nonexistent location"]"
-#define AREACOORD(src) "[src ? "[get_area_name(src, TRUE)] ([src.x], [src.y], [src.z])" : "nonexistent location"]"
+#define COORD(src) "[src ? (src.x ? "([src.x],[src.y],[src.z])" : "held within/by [src.loc]([src.loc.x],[src.loc.y],[src.loc.z])") : "nonexistent location"]"
```
Comment (ShizCalev) [inline code/__DEFINES/admin.dm]: src.loc can also be in nullspace; ie a bomb in a backpack on a person.

Comment (SpaceManiac) [reply code/__DEFINES/admin.dm]: Needs `src.loc?.x` or something like that or else it's gonna runtime if they're in nullspace.
This whole thing may deserve to be a proc if it keeps getting more complicated.

**Review by SpaceManiac — CHANGES_REQUESTED** [review]

File: code/__DEFINES/admin.dm [hunk 2]
```diff
@@ -57,8 +57,8 @@
 #define ADMIN_FULLMONTY_NONAME(user) "[ADMIN_QUE(user)] [ADMIN_PP(user)] [ADMIN_VV(user)] [ADMIN_SM(user)] [ADMIN_FLW(user)] [ADMIN_TP(user)] [ADMIN_INDIVIDUALLOG(user)] [ADMIN_SMITE(user)]"
 #define ADMIN_FULLMONTY(user) "[key_name_admin(user)] [ADMIN_FULLMONTY_NONAME(user)]"
 #define ADMIN_JMP(src) "(<a href='?_src_=holder;[HrefToken(TRUE)];adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)"
-#define COORD(src) "[src ? "([src.x],[src.y],[src.z])" : "nonexistent location"]"
-#define AREACOORD(src) "[src ? "[get_area_name(src, TRUE)] ([src.x], [src.y], [src.z])" : "nonexistent location"]"
+#define COORD(src) "[src ? (src.x ? "([src.x],[src.y],[src.z])" : "held within/by [src.loc]([src.loc.x],[src.loc.y],[src.loc.z])") : "nonexistent location"]"
+#define AREACOORD(src) "[src ? "[get_area_name(src, TRUE)] [COORD(src)]" : "nonexistent location"]"
```
Comment (yoyobatty) [inline code/__DEFINES/admin.dm]: you can probably removed "nonexistant location" part because COORD already takes care of it?

**Comment by MadmanMartian** [comment]
Apologies; I've been caught up in other things.

**Comment by ExcessiveUseOfCobblestone** [comment]
do you want to reopen it?

**Pull request closed without merging**